lib/spack/spack/build_systems/meson.py

```diff
@@ -91,14 +91,13 @@ def std_meson_args(self):
         std_meson_args += getattr(self, 'meson_flag_args', [])
         return std_meson_args
 
-    @staticmethod
-    def _std_args(pkg):
+    def _std_args(self, pkg):
         """Computes the standard meson arguments for a generic package"""
 
         try:
             build_type = pkg.spec.variants['buildtype'].value
         except KeyError:
-            build_type = 'release'
+            build_type = self.build_type.name
```

Comment:
I believe meson is like CMake, and has it's own buildtype variant. That's what this is, we'll have to play nice with it the way we do for CMake.